### PR TITLE
net-mail/gnubiff-2.2.17:  Fix 880267 (clang), 930500 ILICENSE), 933241 (EAPI)

### PR DESCRIPTION
**Pull request by UnixRonin**
Add licenses/GPL-3+-with-openssl-exception
Add net-mail/gnubiff/files/gnubiff-2.2.17-clang.patch to remove -export-dynamic

Closes: https://bugs.gentoo.org/933241
Closes: https://bugs.gentoo.org/880267
Closes: https://bugs.gentoo.org/930500

---

Please check all the boxes that apply:

- [X] I can submit this contribution in agreement with the [Copyright Policy](https://www.gentoo.org/glep/glep-0076.html#certificate-of-origin).
- [X] I have certified the above via adding a `Signed-off-by` line to *every* commit in the pull request.
- [X] This contribution has not been created with the assistance of Natural Language Processing artificial intelligence tools, in accordance with the [AI policy](https://wiki.gentoo.org/wiki/Project:Council/AI_policy).
- [X] I have run `pkgcheck scan --commits --net` to check for issues with my commits.

Please note that all boxes must be checked for the pull request to be merged.

### CODE REVIEW

File: net-mail/gnubiff/gnubiff-2.2.17-r3.ebuild [hunk 1]
```diff
@@ -0,0 +1,47 @@
+# Copyright 1999-2024 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=8
+inherit autotools
+
+DESCRIPTION="A mail notification program"
+HOMEPAGE="http://gnubiff.sourceforge.net/"
```
Comment (mm1ke) [inline net-mail/gnubiff/gnubiff-2.2.17-r3.ebuild:8]:
```suggestion
HOMEPAGE="https://gnubiff.sourceforge.net/"
```

File: net-mail/gnubiff/gnubiff-2.2.17-r3.ebuild [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+	>=gnome-base/libglade-2.3
+	dev-libs/popt
+	password? ( dev-libs/openssl:0= )
+	fam? ( virtual/fam )
+	x11-libs/libX11
+	x11-libs/pango
+	x11-libs/gdk-pixbuf
+"
+DEPEND="${RDEPEND}
+	virtual/pkgconfig
```
Comment (mm1ke) [inline net-mail/gnubiff/gnubiff-2.2.17-r3.ebuild:27]: i guess `virtual/pkgconfig` belongs to `BDEPEND`?

File: net-mail/gnubiff/gnubiff-2.2.17-r3.ebuild [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+RDEPEND="
+	>=x11-libs/gtk+-3:3
+	>=gnome-base/libglade-2.3
+	dev-libs/popt
+	password? ( dev-libs/openssl:0= )
+	fam? ( virtual/fam )
+	x11-libs/libX11
+	x11-libs/pango
+	x11-libs/gdk-pixbuf
+"
```
Comment (mm1ke) [inline net-mail/gnubiff/gnubiff-2.2.17-r3.ebuild:25]: can we sort this please?

```suggestion
RDEPEND="
	dev-libs/popt
	>=gnome-base/libglade-2.3
	x11-libs/gdk-pixbuf
	>=x11-libs/gtk+-3:3
	x11-libs/libX11
	x11-libs/pango
	fam? ( virtual/fam )
	password? ( dev-libs/openssl:0= )
"

File: net-mail/gnubiff/gnubiff-2.2.17-r3.ebuild [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+
+src_configure() {
+	# note: --disable-gnome is to avoid deprecated gnome-panel-2.x
+	econf \
+		--disable-gnome \
+		$(use_enable debug) \
+		$(use_enable nls) \
+		$(use_enable fam) \
+		$(use_with password) \
+		$(use_with password password-string ${RANDOM}${RANDOM}${RANDOM}${RANDOM})
```
Comment (mm1ke) [inline net-mail/gnubiff/gnubiff-2.2.17-r3.ebuild:46]:
```suggestion
	local myeconfargs=(
		--disable-gnome
		$(use_enable debug)
		$(use_enable nls)
		$(use_enable fam)
		$(use_with password)
		$(use_with password password-string ${RANDOM}${RANDOM}${RANDOM}${RANDOM})
	)
	econf "${myeconfargs[@]}"
```

Also, is that something save to set the password-string with all these RANDOM numbers? Just asking.

File: net-mail/gnubiff/Changelog
Comment (mm1ke) [inline net-mail/gnubiff/Changelog:1]: Changelog files were used before the git migration. I guess this should be removed.

File: net-mail/gnubiff/gnubiff-2.2.17-r3.ebuild [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+EAPI=8
+inherit autotools
+
+DESCRIPTION="A mail notification program"
+HOMEPAGE="http://gnubiff.sourceforge.net/"
+SRC_URI="https://downloads.sourceforge.net/${PN}/${P}.tar.gz"
+
+LICENSE="GPL-3+-with-openssl-exception"
+SLOT="0"
+KEYWORDS="amd64 x86"
```
Comment (mm1ke) [inline net-mail/gnubiff/gnubiff-2.2.17-r3.ebuild:13]:
```suggestion
KEYWORDS="~amd64 ~x86"
```
You cannot commit straight to stable.

File: net-mail/gnubiff/gnubiff-2.2.17-r3.ebuild (hunk 4 above)
Comment (UnixRonin) [reply net-mail/gnubiff/gnubiff-2.2.17-r3.ebuild:46]: I have no idea what that's for, this isn't my code, I'm just trying to give it an extra lease on life.

File: net-mail/gnubiff/gnubiff-2.2.17-r3.ebuild (hunk 3 above)
Comment (UnixRonin) [reply net-mail/gnubiff/gnubiff-2.2.17-r3.ebuild:25]: I admit I'm more concerned about the warning that libglade is deprecated, but without going in and re-engineering the code I don't see a fix for that.

File: net-mail/gnubiff/gnubiff-2.2.17-r3.ebuild [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+	fam? ( virtual/fam )
+	x11-libs/libX11
+	x11-libs/pango
+	x11-libs/gdk-pixbuf
+"
+DEPEND="${RDEPEND}
+	virtual/pkgconfig
+"
+
+DOCS="AUTHORS ChangeLog NEWS README THANKS TODO"
```
[the remark this answers is not in the document]
Comment (UnixRonin) [reply net-mail/gnubiff/gnubiff-2.2.17-r3.ebuild:30]: I'll take a look.

File: net-mail/gnubiff/gnubiff-2.2.17-r3.ebuild (hunk 2 above)
Comment (UnixRonin) [reply net-mail/gnubiff/gnubiff-2.2.17-r3.ebuild:27]: Fixed that too, it's cleaner

File: net-mail/gnubiff/gnubiff-2.2.17-r3.ebuild (hunk 6 above)
[the remark this answers is not in the document]
Comment (UnixRonin) [reply net-mail/gnubiff/gnubiff-2.2.17-r3.ebuild]: Verified it's unnecessary.

File: net-mail/gnubiff/gnubiff-2.2.17-r3.ebuild [hunk 7]
```diff
@@ -0,0 +1,47 @@
...
+	>=gnome-base/libglade-2.3
+	x11-libs/gdk-pixbuf
+	>=x11-libs/gtk+-3:3
+	x11-libs/libX11
+	x11-libs/pango
+	fam? ( virtual/fam )
+	password? ( dev-libs/openssl:0= )
+"
+BDEPEND="
+	virtual/pkgconfig
```
Comment (mm1ke) [inline net-mail/gnubiff/gnubiff-2.2.17-r3.ebuild:27]: I think you accidentally removed the `DEPEND` part? This still should be valid:

```suggestion
"
DEPEND="${RDEPEND}"
BDEPEND="
	virtual/pkgconfig
```

Comment (UnixRonin) [reply net-mail/gnubiff/gnubiff-2.2.17-r3.ebuild:27]: I misunderstood.  My fault.

**Comment by thesamesam** [comment]
Thanks. Please file an upstream bug for the use of deprecated libglade and include a link to it in the ebuild (and include it in the commit message). Then squash the commits here and it's good to go, I think!

File: net-mail/gnubiff/Changelog [hunk 8]
```diff
@@ -0,0 +1,11 @@
+# Changelog for net-mail/gnubiff
```
Comment (thesamesam) [inline net-mail/gnubiff/Changelog:1]: We don't do these anymore -- could you show me which document told you to do it? (It's fine if you just made a mistake but I'm wondering if there's some old doc giving bad instructions.)

Comment (UnixRonin) [reply net-mail/gnubiff/Changelog:1]: I think I got the habit from looking at other ebuilds.  If we no longer do that, I'll make a note to stop creating them.

Comment (thesamesam) [reply net-mail/gnubiff/Changelog:1]: You shouldn't see any ChangeLogs in ::gentoo since like 2015 :)

**Comment by UnixRonin** [comment]
> Thanks. Please file an upstream bug for the use of deprecated libglade and include a link to it in the ebuild (and include it in the commit message). Then squash the commits here and it's good to go, I think!

Sam, I think through git ineptitude I've managed to completely screw this up and my PR has acquired over 2,000 extra commits.  Is there a way to clean it up or should I just delete the entire PR and start over?  (And if so, how do I do so without screwing it up a second time?)

**Comment by AngryLoki** [comment]
> Is there a way to clean it up

Something like this
```
git checkout master
git pull
git checkout gnubiff-2.2.17-r3
git reset --hard master
git cherry-pick a249c0a3d48dece283062ff580dc6d6b24a7207a bbdb0d0525bcdfefbed37fb9ab0c1ccc52faa4af fd5aa99237fb80db3fa5a12b13b65193f81a5fb6 0433db9c23c363a181a7ded780c109c212e8b916 d34ac800fa8f8f73bb329c91dfcd867d3d6ec367 73c492c9652c7522e61be30fb3b579c429ee9bf9 b0f9cc05401900bd33ca65e811d058509fb3f121
git push --force-with-lease
```

**Comment by eli-schwartz** [comment]
> > Is there a way to clean it up
> 
> Something like this
> 
> ```
> git checkout master
> git pull
> git checkout gnubiff-2.2.17-r3
> git reset --hard master
> ```

This can be simplified to:
```
git fetch origin
git reset --hard origin/master
```

**Comment by eli-schwartz** [comment]
Changes made by this PR:
```diff
diff --git a/net-mail/gnubiff/gnubiff-2.2.17-r2.ebuild b/net-mail/gnubiff/gnubiff-2.2.17-r3.ebuild
similarity index 57%
copy from net-mail/gnubiff/gnubiff-2.2.17-r2.ebuild
copy to net-mail/gnubiff/gnubiff-2.2.17-r3.ebuild
index 3ad5e5bfb620..296b36941113 100644
--- a/net-mail/gnubiff/gnubiff-2.2.17-r2.ebuild
+++ b/net-mail/gnubiff/gnubiff-2.2.17-r3.ebuild
@@ -1,49 +1,52 @@
 # Copyright 1999-2024 Gentoo Authors
 # Distributed under the terms of the GNU General Public License v2
+#
+# This package depends upon deprecated gnome-base/libglade.  An
+# upstream but has been filed on this issue.
+# (https://sourceforge.net/p/gnubiff/bugs/67/)
 
-EAPI=6
+EAPI=8
 inherit autotools
 
 DESCRIPTION="A mail notification program"
-HOMEPAGE="http://gnubiff.sourceforge.net/"
+HOMEPAGE="https://gnubiff.sourceforge.net/"
 SRC_URI="https://downloads.sourceforge.net/${PN}/${P}.tar.gz"
 
-LICENSE="GPL-3"
+LICENSE="GPL-3+-with-openssl-exception"
 SLOT="0"
-KEYWORDS="amd64 x86"
+KEYWORDS="~amd64 ~x86"
 IUSE="debug fam nls password"
 
 RDEPEND="
-	>=x11-libs/gtk+-3:3
-	>=gnome-base/libglade-2.3
 	dev-libs/popt
-	password? ( dev-libs/openssl:0= )
-	fam? ( virtual/fam )
+	>=gnome-base/libglade-2.3
+	x11-libs/gdk-pixbuf
+	>=x11-libs/gtk+-3:3
 	x11-libs/libX11
 	x11-libs/pango
-	x11-libs/gdk-pixbuf
+	fam? ( virtual/fam )
+	password? ( dev-libs/openssl:0= )
 "
-DEPEND="${RDEPEND}
+DEPEND="${RDEPEND}"
+BDEPEND="
 	virtual/pkgconfig
 "
 
-DOCS="AUTHORS ChangeLog NEWS README THANKS TODO"
-
 src_prepare() {
-	eapply -p0 "${FILESDIR}/${PN}-2.2.15-fix-nls.patch"
-	eapply -p1 "${FILESDIR}/${PN}-2.2.15-gold.patch"
-	eapply -p0 "${FILESDIR}/${PN}-2.2.15-underlink.patch"
-	eautoreconf
+	eapply "${FILESDIR}/${PN}-2.2.17-clang.patch"
 	eapply_user
+	eautoreconf
 }
 
 src_configure() {
 	# note: --disable-gnome is to avoid deprecated gnome-panel-2.x
-	econf \
-		--disable-gnome \
-		$(use_enable debug) \
-		$(use_enable nls) \
-		$(use_enable fam) \
-		$(use_with password) \
+	local myeconfargs=(
+		--disable-gnome
+		$(use_enable debug)
+		$(use_enable nls)
+		$(use_enable fam)
+		$(use_with password)
 		$(use_with password password-string ${RANDOM}${RANDOM}${RANDOM}${RANDOM})
+	)
+	econf "${myeconfargs[@]}"
 }
```

**Comment by eli-schwartz** [comment]
Dropping the existing eapply'ed patches looks like it's probably an accident.

**Pull request closed without merging**